cmd/main.go

```diff
@@ -2,57 +2,101 @@ package main
 
 import (
 	"context"
+	"flag"
+	"fmt"
 	"log/slog"
 	"net/http"
 	"os"
 
 	"github.com/grafana/crocochrome"
 	crocohttp "github.com/grafana/crocochrome/http"
+	"github.com/grafana/crocochrome/internal/version"
+
 	"github.com/grafana/crocochrome/metrics"
 	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
+type Config struct {
+	UserGroup int
+	TempDir   string
+}
+
 func main() {
 	logger := slog.New(slog.NewJSONHandler(os.Stderr, &slog.HandlerOptions{
 		Level: slog.LevelDebug,
 	}))
 
+	config := &Config{}
+	flag.StringVar(&config.TempDir, "temp-dir", "/chromium-tmp", "Directory for chromiumium instances to write their data to")
+	flag.IntVar(&config.UserGroup, "user-group", 65534, "Default user to run as. For local development, set this flag to 0")
+
+	flag.Parse()
+	if err := run(logger, config); err != nil {
+		logger.Error("run failed to execute",
+			slog.String("msg", err.Error()))
+	}
+}
+
+func run(logger *slog.Logger, config *Config) error {
+	logger.LogAttrs(context.Background(), slog.LevelInfo, "Starting crocochrome supervisor",
+		slog.String("version", version.Short()),
+		slog.String("commit", version.Commit()),
+		slog.String("timestamp", version.Buildstamp()),
+	)
+
 	mux := http.NewServeMux()
 
 	registry := prometheus.NewRegistry()
 
+	registry.MustRegister(prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: "sm",
+			Subsystem: "crocochrome",
+			Name:      "info",
+			Help:      "Crocochrome Info",
+			ConstLabels: prometheus.Labels{
+				"version":   version.Short(),
+				"commit":    version.Commit(),
+				"timestamp": version.Buildstamp(),
+			},
+		},
+	))
```

Comment:
We do have a `metrics` package that tries to encapsulate `MustRegister` and the likes. For a metric that will never change is not a big deal, but if we move this there we can access the constants for `Namespace`, `Subsystem`, etc. WDYT about that?